src/core/thread/router_table.hpp

```diff
@@ -265,6 +265,20 @@ class RouterTable : public InstanceLocator
      */
     Router *GetRouter(const Mac::ExtAddress &aExtAddress);
 
+    /**
+     * This method returns if the router table contains a given `Neighbor` instance.
+     *
+     * @param[in]  aNeighbor  A reference to a `Neighbor`.
+     *
+     * @returns Whether the router table contains @p aNeighbor.
+     *
+     */
+    bool Contains(const Neighbor &aNeighbor) const
+    {
+        return mRouters <= &static_cast<const Router &>(aNeighbor) &&
+               &static_cast<const Router &>(aNeighbor) < mRouters + Mle::kMaxRouters;
```

Comment:
It seems to me always return True. When would it be false? @simonlingoogle 

```
return mRouters <= &static_cast<const Router &>(aNeighbor) &&
               &static_cast<const Router &>(aNeighbor) < mRouters + Mle::kMaxRouters;
```

Reply:
when aNeighbor is not a Router in mRouters, e.x. mParent, mParentCandidate

Reply:
how about adding note to illustrate False scenarios for easier readability?

Reply:
comment added